src/components/cc-badge/cc-badge.js

```diff
@@ -36,35 +39,69 @@ export class CcBadge extends LitElement {
     /** @type {BadgeIntent} Sets the accent color used for the badge. */
     this.intent = 'neutral';
 
+    /** @type {boolean} Whether the component should be displayed as skeleton. */
+    this.skeleton = false;
+
     /** @type {BadgeWeight} Sets the style of the badge depending on how much one wants it to stand out. */
     this.weight = 'dimmed';
   }
 
   render () {
+    const modes = {
+      dimmed: this.weight == null || this.weight === 'dimmed',
+      strong: this.weight === 'strong',
+      outlined: this.weight === 'outlined',
+      neutral: this.intent == null || this.intent === 'neutral',
+      info: this.intent === 'info',
+      success: this.intent === 'success',
+      warning: this.intent === 'warning',
+      danger: this.intent === 'danger',
+      skeleton: this.skeleton,
+      circle: this.circle,
+    };
+
     return html`
-      ${this.iconSrc != null ? html`
-        <img src=${this.iconSrc} alt=${this.iconAlt ?? ''}>
-      ` : ''}
-      <span>
+      <span class="${classMap(modes)}">
+        ${this.iconSrc != null ? html`
+          <img src=${this.iconSrc} alt=${this.iconAlt ?? ''}>
+        ` : ''}
         <slot></slot>
```

Comment:
you need this <span> around the <slot> because we are using inline-flex.
If we slot something like my super <strong>slotted text</strong> then there will be a flex gap between the text node and the <strong>.

It might also cause other issues which I don't remember (I used to do this a lot before @hsablonniere reviews smile